### PR TITLE
build: do not rely on CommonJS module global for Bazel karma saucelabs

### PR DESCRIPTION
**Pull request by devversion**
The Karma Saucelabs script for Bazel & Saucelabs relies on some CommonJS
specific features. This commit replaces it with an ESM-compatible
alternative so that it can execute because `nodejs_binary` requires ESM
files now.

### CODE REVIEW

File: tools/saucelabs/karma-saucelabs.mjs [hunk 1]
```diff
@@ -65,10 +71,20 @@ try {
...
+  await launchNodeBinaryAndWait(karmaBin, process.argv.slice(2));
+}
+
+function launchNodeBinaryAndWait(entryPointPath, args) {
+  return new Promise((resolve, reject) => {
+    const proc = childProcess.fork(karmaBin, args, {stdio: 'inherit'});
+    proc.on('error', (err) => reject(err));
+    proc.on('close', (code) =>
+      code !== 0
+        ? reject(new Error(`Script "${entryPointPath}" exited with non-zero status code`))
```
Comment (JoostK) [inline tools/saucelabs/karma-saucelabs.mjs:84]:
```suggestion
        ? reject(new Error(`Script "${entryPointPath}" exited with non-zero status code ${code}`))
```

File: tools/saucelabs/karma-saucelabs.mjs [hunk 2]
```diff
@@ -65,10 +71,20 @@ try {
...
-} catch (e) {
-  console.error(e.stack || e);
-  process.exit(1);
+
+  await launchNodeBinaryAndWait(karmaBin, process.argv.slice(2));
+}
+
+function launchNodeBinaryAndWait(entryPointPath, args) {
+  return new Promise((resolve, reject) => {
+    const proc = childProcess.fork(karmaBin, args, {stdio: 'inherit'});
```
Comment (JoostK) [inline tools/saucelabs/karma-saucelabs.mjs:80]:
```suggestion
    const proc = childProcess.fork(entryPointPath, args, {stdio: 'inherit'});
```

**Review by JoostK — APPROVED** [review]
Some nits but otherwise LGTM

**Comment by devversion** [comment]
This PR was merged into the repository by commit 74ba3021da24e1cec05bcd60df10c5a9c9f8c0c0.

**Pull request closed without merging**